demo/checkbox-validation-demos.html

```diff
@@ -0,0 +1,40 @@
+<dom-module id="checkbox-validation-demos">
+  <template>
+    <style include="vaadin-component-demo-shared-styles">
+      :host {
+        display: block;
+      }
+    </style>
+
+    <h3>Required</h3>
+    <vaadin-demo-snippet id="checkbox-required" when-defined="vaadin-checkbox-group">
+      <template preserve-content>
+        <vaadin-checkbox-group label="Required" required error-message="Please choose one option" theme="vertical"
+          id="checkboxGroup">
+          <vaadin-checkbox value="1">Option one</vaadin-checkbox>
+          <vaadin-checkbox value="2">Option two</vaadin-checkbox>
+          <vaadin-checkbox value="3">Option three</vaadin-checkbox>
+        </vaadin-checkbox-group>
+        <vaadin-button id="submit">Submit</vaadin-button>
+        <script>
+          window.addDemoReadyListener('#checkbox-required', function(document) {
+            var submit = document.getElementById('submit');
```

Comment:
nit: We could align these better with the other demos by changing:
`var` -> `const`
`getElementById` -> `querySelector`

Reply:
Done.